streamer/pulp/streamer/server.py

```diff
@@ -174,84 +113,183 @@ def _handle_get(self, request):
         Download the requested content using the content unit catalog and dispatch
         a celery task that causes Pulp to download the newly cached unit.
 
-        :param request: The content request.
+        :param request: An HTTP request.
         :type  request: twisted.web.server.Request
         """
-        catalog_path = urlparse(request.uri).path
         with Responder(request) as responder:
             try:
-                catalog_entry = model.LazyCatalogEntry.objects(
-                    path=catalog_path).order_by('importer_id').first()
-                if not catalog_entry:
-                    raise DoesNotExist()
-                self._download(catalog_entry, request, responder)
-            except DoesNotExist:
-                logger.error(_('Failed to find a catalog entry with path'
-                               ' "{rel}".'.format(rel=catalog_path)))
-                request.setResponseCode(NOT_FOUND)
-            except PluginNotFound:
-                msg = _('Catalog entry for {rel} references a plugin id'
-                        ' which is not valid.')
-                logger.error(msg.format(rel=catalog_path))
-                request.setResponseCode(INTERNAL_SERVER_ERROR)
+                path = urlparse(request.uri).path
+                q_set = LazyCatalogEntry.objects.filter(path=path)
+                q_set = q_set.order_by('-_id', '-revision')
+                count = q_set.count()
+                if not count:
+                    logger.error(_('No catalog entry found. path={p}'.format(p=path)))
+                    request.setResponseCode(NOT_FOUND)
+                    return
+                for entry in q_set.all():
+                    logger.info('Trying URL: {url}'.format(url=entry.url))
```

Comment:
I understand where you're coming from.   But, based on past experience, getting debug level logging output from customers can be difficult.  On EL6 and probably EL7, it requires users to reconfigure rsyslog and then recreate the failure.  After spending a lot of time struggling to customer cases involving lazy, I think it's better to leave this at `info` for now.